source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,206 @@
+ pr₁ (wlem-lemma i P) r p = nequal i (≤-is-antisymmetric A (order i) (pr₁ r (inr p)))
+ pr₂ (wlem-lemma i P) r np = nequal i (≤-is-antisymmetric A (order i) (pr₂ r (( lower i , h ))))
+  where
+   h : ((lower i) is-an-upper-bound-of ((𝟙 + P holds) , δ (lower i) (upper i) P)) holds
+   h (inl ✯) = ≤-is-reflexive A (lower i)
+   h (inr p) = 𝟘-induction (np p)
+    
+\end{code}
+
+We now show that the two element poset is δ_𝓥 complete only if WLEM holds.
+
+\begin{code}
+
+2-is-non-trivial : non-trivial-posets.is-non-trivial-poset 2-Poset
+2-is-non-trivial = (₀ , ₁ , ⋆ , zero-is-not-one)
+
+2-is-δ_𝓥-complete-WLEM : {𝓥 : Universe} →
+ δ_𝓥-complete-poset.is-δ_𝓥-complete {𝓤₀} {𝓤₀} 𝓥 2-Poset →
+ (P : Ω 𝓥) → is-decidable (¬ (P holds))
+2-is-δ_𝓥-complete-WLEM {𝓥} i P = decide-¬P
+ where
+  open Joins (rel-syntax 2-Poset)
+  open δ_𝓥-complete-poset 𝓥 2-Poset
+  open non-trivial-posets 2-Poset
+
+  h : Σ s ꞉ ∣ 2-Poset ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ ₀ ₁ P))) holds
+  h = i ₀  ₁ ⋆ P
+
+  g : Σ s ꞉ ∣ 2-Poset ∣ₚ , (s is-lub-of ((𝟙 + P holds) , (δ ₀ ₁ P))) holds →
+    is-decidable (¬ (P holds))
+  g (₀ , sup) = inl (pr₁ (wlem-lemma 2-is-non-trivial P) sup)
+  g (₁ , sup) = inr (pr₂ (wlem-lemma 2-is-non-trivial P) sup)
+
+  decide-¬P : is-decidable (¬ (P holds))
+  decide-¬P = g h
+
+\end{code}
+
+Since non-trivial is a negated concept it only has enough strength to derive WLEM, we now introduce the stronger concept of positivity.
+
+\begin{code}
+
+module positive-posets {𝓤  𝓦  𝓥 : Universe} (A : Poset 𝓤 𝓦) where
+ open Joins (rel-syntax A)
+ open δ_𝓥-complete-poset 𝓥 A
+ open Universal fe
+ open PosetReasoning A
+
+ _is-strictly-below_ : (x y : ∣ A ∣ₚ) → {i : is-δ_𝓥-complete} → 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+ _is-strictly-below_ x y = (x ≤[ A ] y) holds ×
+  ((z : ∣ A ∣ₚ) → (y ≤[ A ] z) holds → (P : Ω 𝓥) →
+  (z is-lub-of ((𝟙 + P holds) , δ x z P)) holds → P holds)
+
+ order : {x y : ∣ A ∣ₚ} → x is-strictly-below y → (x ≤[ A ] y) holds
+ order c = pr₁ c
+
+ sup-condition : {x y : ∣ A ∣ₚ} → x is-strictly-below y →
+  ((z : ∣ A ∣ₚ) → (y ≤[ A ] z) holds → (P : Ω 𝓥) →
+  (z is-lub-of ((𝟙 + P holds) , δ x z P)) holds → P holds)
+ sup-condition c = pr₂ c
+
+ strictly-below-implies-non-trivial : (x y : ∣ A ∣ₚ) → is-δ_𝓥-complete → (x is-strictly-below y)
+  → (x ≤[ A ] y) holds × ¬ (x ＝ y)
+ pr₁ (strictly-below-implies-non-trivial x y i c) = order c
+ pr₂ (strictly-below-implies-non-trivial x y i c) p =
+  𝟘-induction (sup-condition c y (≤-is-reflexive A y) ((𝟘{𝓥} , 𝟘-is-prop {𝓥})) ((g , h)))
+   where
+    g : (y is-an-upper-bound-of ((𝟙 + ((𝟘 , 𝟘-is-prop) holds)) , δ x y (𝟘 , 𝟘-is-prop))) holds
+    g (inl ⋆) = order c
+
+    h : (Ɐ u ꞉ (upper-bound ((𝟙 + ((𝟘 , 𝟘-is-prop) holds)) , δ x y (𝟘 , 𝟘-is-prop))) ,
+     y ≤[ A ] (pr₁ u)) holds
+    h u = y ＝⟨ p ⁻¹ ⟩ₚ pr₂ u (inl ⋆)
+
+\end{code}
+
+We could show that if the converse holds then so does LEM in 𝓥.
+
+\begin{code}
+
+ transitivity-lemma : (x y z : ∣ A ∣ₚ) → is-δ_𝓥-complete →
```

Comment:
You should split this into two lemmas. In formalizing, it's often convenient to have small units, rather than big lemmas that will later need projections to be useful.